src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -189,24 +211,37 @@ public static InnerSafeCloseSocket CreateSocket(AddressFamily addressFamily, Soc
                 int pt = (int)protocolType;
 
                 int fd = Interop.libc.socket(af, sock, pt);
+                if (fd != -1)
+                {
+                    // The socket was created successfully; make it non-blocking.
+                    int err = Interop.Sys.Fcntl.SetIsNonBlocking(fd, 1);
+                    if (err != 0)
+                    {
+                        Interop.Sys.Close(fd);
+                        fd = -1;
+                    }
+                }
```

Comment:
I'm probably just missing it, but where are we setting IsNonBlocking on the SafeHandle?